src/SymbolProvider/include/SymbolProvider/SymbolProvider.h

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef SYMBOL_PROVIDER_SYMBOL_PROVIDER_H_
+#define SYMBOL_PROVIDER_SYMBOL_PROVIDER_H_
+
+#include "OrbitBase/CanceledOr.h"
+#include "OrbitBase/Result.h"
+#include "OrbitBase/StopToken.h"
+#include "SymbolProvider/ModuleIdentifier.h"
+
+namespace orbit_symbol_provider {
+
+// "SymbolProvider" supports retrieving symbols for the provided module from its symbol source, and
+// provides the local file path where symbols are cached if retrieval succeeded.
+// It also allows to query whether module symbols can be found in its symbol source.
+class SymbolProvider {
+ public:
+  virtual ~SymbolProvider() = default;
+
+  // Search for symbols for the provided module in the SymbolProvider's symbol source. Return an
+  // ErrorMessage if error occurs or symbols are not found, void otherwise.
+  [[nodiscard]] virtual orbit_base::Future<ErrorMessageOr<void>> FindSymbols(
+      const ModuleIdentifier& module_id) = 0;
```

Comment:
Do we also want to have a stop token here? @antonrohr

Reply:
I dont think we need it, but I am not sure. I would say lets leave it as, we can still add one in the future.

Reply:
Henning and I just had a offline discussion. Having a retrieve might be enough for our current use case.  
I'm going to remove the `FindSymbols` method now. We can add it later if it's needed in the future.
Thanks you both for the help! : )